docs/import/manage_disaster_base.js

```diff
@@ -0,0 +1,805 @@
+let scoreBoundsMap;
+
+//                       Damage-related functions
+
+const SCORE_COORDINATES_PATH = Object.freeze(['scoreBoundsCoordinates']);
+const DAMAGE_PROPERTY_PATH = Object.freeze(['damageAssetPath']);
+
+/**
+ * Does all initialization for damage asset and related fields. Creates damage
+ * select, no-damage column and value, puts everything in a pending state but
+ * with initial visibility based on Firestore values, and when all data ready,
+ * finishes display.
+ * @param {AssetData} assetData
+ * @return {Promise<void>}
+ */
+async function initializeDamage(assetData) {
+  startPending();
+  if (isFlexible()) {
+    initializeScoreBoundsMapFromAssetData(assetData);
+  } else {
+    initializeScoreBoundsMapFromAssetData(
+        assetData, assetData.stateBasedData.states);
+  }
+  const damageIntroSpan = $('#damage-intro-span');
+  const damageDiv = $('#damage-asset-div').empty().append(damageIntroSpan);
+  const damageSelect = createSelect(DAMAGE_PROPERTY_PATH)
+                           .on('change',
+                               () => displayDamageRelatedElements(
+                                   startPendingWriteSelectAndGetPropertyNames(
+                                       DAMAGE_PROPERTY_PATH),
+                                   damageSelect.val()));
+  damageDiv.append(damageSelect);
+  createNoDamageColumnAndValueList();
+  showHideDamageAndMapDivs(!!getStoredValueFromPath(DAMAGE_PROPERTY_PATH));
+  if (!await getAssetsAndSetOptionsForSelect(DAMAGE_PROPERTY_PATH, false)) {
+    return;
+  }
+  return displayDamageRelatedElements(
+      verifyAsset(DAMAGE_PROPERTY_PATH, []), damageSelect.val());
+}
+
+/** @type {ColumnInfo} */
+const NODAMAGE_COLUMN_INFO = {
+  label: 'column that can distinguish between damaged and undamaged buildings',
+  // Actual text on page modified when flexible using setExplanationSpanText.
+  explanation: 'optional',
+  path: ['noDamageKey'],
+};
+
+/** @type {ColumnInfo} */
+const NODAMAGE_VALUE_INFO = {
+  label: 'value in column that identifies undamaged buildings',
+  // Actual text on page modified when flexible using setExplanationSpanText.
+  explanation: 'required if column is set',
+  path: ['noDamageValue'],
+};
+
+/**
+ * Displays all damage-related elements based on current value of damage asset
+ * select. Invoked both during initialization and on damage asset change. Will
+ * put no-damage column into a "pending" state until columns available.
+ * @param {Promise<?Array<EeColumn>>} propertyNamesPromise Promise that will
+ *     contain columns of damage asset. Created via {@link verifyAsset} or
+ *     {@link startPendingWriteSelectAndGetPropertyNames}
+ * @param {?EeFC} damageAsset Value of damage asset, from page
+ * @return {Promise<void>}
+ */
+async function displayDamageRelatedElements(propertyNamesPromise, damageAsset) {
+  setNoDamageColumnAndValue(null);
+  showHideDamageAndMapDivs(!!damageAsset);
+  try {
+    const propertyNames = await propertyNamesPromise;
```

Comment:
No, but hard to guarantee, since we're calling out to both Firestore and EE.

If you're asking because of the `try {}` block, it's helpful to also execute the `finally` clause on early returns.